activiti-core/activiti-api-impl/activiti-api-runtime-shared-impl/src/main/java/org/activiti/runtime/api/event/impl/ToVariableUpdatedConverter.java

```diff
@@ -38,6 +39,10 @@ public Optional<VariableUpdatedEvent> from(ActivitiVariableUpdatedEvent internal
 
         VariableInstanceImpl<Object> variableInstance = createVariableInstance(internalEvent, isEphemeral);
 
+        if(isEphemeral) {
+            ((ActivitiVariableUpdatedEventImpl) internalEvent).setVariablePreviousValue(null);
+        }
+
         return Optional.of(new VariableUpdatedEventImpl<>(variableInstance, internalEvent.getVariablePreviousValue()));
```

Comment:
I think it's better to use temporary variable to store the value to be passed as previous value, this will avoid modifying the original internal event and also avoid the usage of casting.

```suggestion
       Object previousValue = isEphemeral? null : internalEvent.getVariablePreviousValue();
        return Optional.of(new VariableUpdatedEventImpl<>(variableInstance, previousValue));
```